### ISSUE TITLE
Supported HDD Partitioning Utilities?

### ISSUE THREAD

**Issue by richard-gravy** [dreamer/dosbox-staging#252]
Hello,

I am trying to create a new HDD image for storage of DOS apps, using a single FAT16 partition.
After some disagreement between myself and IMGMOUNT as to how many cylinders my raw image has, I went down the documented route for HDD images and created it using Bochs's bximage utility.
This told me that cylinders = 1320.
I was then able to mount the image with:-
`imgmount 2 /tmp/hda.img -size 512,63,16,1320 -fs none`
The usual procedure for me at this time would be to use FDISK to create a single primary DOS partition, but there were two problems:-
I was using FDISK from MS-DOS 6.22, which checks the DOS version.  This was resolved by setting the reported version:-
`ver set 6.22`
Then, after attempting to create the partition, Dosbox-staging aborts with the message:-
`Exit to error: DOS:0x65:Unhandled country information call 23`

I know why this isn't implemented, and it's a bit odd that FDISK needs country information to set a partition on the MBR.  But is there a known working utility that I can use for this?

**Comment by richard-gravy** [comment]
I think I've solved this one through a bit more trial and error.
_Note to self: trying to use HDD images larger than 500MB will spook the horses._

**Comment by dreamer** [comment]
Just in case someone will find this bug report in the future.

Creating a HDD image for use inside dosbox is a painful experience. Long time ago, there was a DOSBox version called "HAL Megabuild" (I learned about it from [here](https://docs.google.com/document/d/1hvgFvAYjPG93h-Avun3sprvZX2GfkRhl4YJBT15FTx0/edit#)), which included built-in utility for easy creation of images, which was ported from [Bochs](http://bochs.sourceforge.net/) emulator.

This utility never made it into upstream project - I don't know why.

I don't plan to work on this at this time, but if anyone wants to pick it up - I managed to recover source code of HAL's Megabuild from internet archive and imported it to a branch `vogons/hal9k-megabuild6-r3661`, as this commit: 2ce05a79bbb9af7023d614e7a6ca7770b46e3104, so this is a good place to start.
